src/components/CompareResults/ResultsMain.tsx

```diff
@@ -61,13 +67,21 @@ function ResultsMain() {
     setSearchParams(searchParams);
   };
 
+  const onSearchTermChange = (newSearchTerm: string) => {
+    setSearchTerm(newSearchTerm);
+    updateSearchParams('search', newSearchTerm);
```

Comment:
Yeah, I'd use a "replaceState".

My rationale is that when the user presses the back button, IMO they expect to go to the previous page, not the previous search. They don't especially have the mental model that changing the search will push a new history item, because that's not what happens in most sites I know.

The profiler is different because it's more like an "app" and we expect to be able to undo things. Although in the profiler we could possibly use replaceState for that too for the same reason, because searching is not really an action.